server/src/hardware/protocol/z21/messages.hpp

```diff
@@ -1251,16 +1292,24 @@ struct LanXLocoInfo : LanX
       else
         f21f28 &= ~flag;
     }
+    else if(index <= 31 && supportsF29F31())
+    {
+      const uint8_t flag = (1 << (index - 29));
+      if(value)
+        f29f31 |= flag;
+      else
+        f29f31 &= ~flag;
+    }
   }
 
-  void calcChecksum()
+  inline void updateChecksum()
   {
-    checksum = xheader;
-    for(uint8_t* db = &addressHigh; db < &checksum; db++)
-      checksum ^= *db;
+    //Data length - 7 Z21 header bytes + 1 byte for db0
+    LanX::updateChecksum(dataLen() - 6);
   }
 } ATTRIBUTE_PACKED;
-static_assert(sizeof(LanXLocoInfo) == 14);
+static_assert(sizeof(LanXLocoInfo) >= LanXLocoInfo::minMessageSize &&
+              sizeof(LanXLocoInfo) <= LanXLocoInfo::maxMessageSize);
```

Comment:
size of this struct is now always 15, the `static_assert` is only checked at compile time

Reply:
Isn't it cool to have an abstract compile time logic? So it does not have to be changed when we change the struct layout

Reply:
I don't know if we already reched max size, using every byte field in the struct. Maybe there is still reserved space for future expansion.